Missing symbols when building static on OSX

I'm observing missing symbols on my OSX platform when building via cmake with `-DBUILD_SHARED_LIBS=OFF`.  Will update with more details as I debug; it's entirely possible that these are not limited to OSX, which reveals another yet another blind spot in our CI testing, which I will patch as part of fixing this.  Related to (or at least discovered) when fixing #1829.

First example:

````
[ 27%] Linking C executable ncdumpchunks
Undefined symbols for architecture x86_64:
  "_zutest", referenced from:
      _NCZ_transfer in libnetcdf.a(zwalk.c.o)
      _NCZ_walk in libnetcdf.a(zwalk.c.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [ncgen/ncdumpchunks] Error 1
````

--- a/libnczarr/zdebug.c
+++ b/libnczarr/zdebug.c
@@ -11,6 +11,13 @@
 /* Mnemonic */
 #define RAW 1
 
+/**************************************************/
+/* Data Structure  */
+
+struct ZUTEST* zutest = NULL;
+
+/**************************************************/
+
 #ifdef ZCATCH
 /* Place breakpoint here to catch errors close to where they occur*/
 int
@@ -32,13 +39,6 @@ zthrow(int err, const char* file, int line)
 #endif
     return zbreakpoint(err);
 }
-#endif
-
-/**************************************************/
-/* Data Structure  */
-
-#ifdef ZUT
-struct ZUTEST zutest;
 #endif
 
 /**************************************************/

--- a/libnczarr/zdebug.h
+++ b/libnczarr/zdebug.h
@@ -5,9 +5,6 @@
 #ifndef ZDEBUG_H
 #define ZDEBUG_H
 
-/* Enable unit testing; on by default */
-#define ZUT
-
 #undef ZDEBUG /* general debug */
 #undef ZDEBUG1 /* detailed debug */
 
@@ -65,12 +62,11 @@ EXTERNL void NCZbacktrace(void);
 #define UTEST_WALK	2
 #define UTEST_TRANSFER	4
 
-#ifdef ZUT
-EXTERNL struct ZUTEST {
+struct ZUTEST {
     int tests;
     void (*print)(int sort,...);
-} zutest;
-#endif
+};
+EXTERNL struct ZUTEST* zutest;
 
 #endif /*ZDEBUG_H*/
 

--- a/libnczarr/zwalk.c
+++ b/libnczarr/zwalk.c
@@ -172,10 +172,8 @@ fprintf(stderr,"allprojections:\n%s",nczprint_allsliceprojections(common->rank,c
 	    slpslices[r] = proj[r]->chunkslice;
 	    memslices[r] = proj[r]->memslice;
 	}
-#ifdef ZUT
-	if(zutest.tests & UTEST_TRANSFER)
-	    zutest.print(UTEST_TRANSFER, common, chunkodom, slpslices, memslices);
-#endif
+	if(zutest && zutest->tests & UTEST_TRANSFER)
+	    zutest->print(UTEST_TRANSFER, common, chunkodom, slpslices, memslices);
 
 	slpodom = nczodom_fromslices(common->rank,slpslices);
 	memodom = nczodom_fromslices(common->rank,memslices);
@@ -251,10 +249,8 @@ fflush(stderr);
 #endif
 
 	    LOG((1,"%s: slpptr0=%p memptr0=%p slpoffset=%llu memoffset=%lld",__func__,slpptr0,memptr0,slpoffset,memoffset));
-#ifdef ZUT
-	    if(zutest.tests & UTEST_WALK)
-		zutest.print(UTEST_WALK, common, chunkodom, slpodom, memodom);
-#endif
+	    if(zutest && zutest->tests & UTEST_WALK)
+		zutest->print(UTEST_WALK, common, chunkodom, slpodom, memodom);
 	    if((stat = transfern(slpodom,memodom,common,slpptr0,memptr0))) goto done;
             nczodom_next(memodom);
         } else break; /* slpodom exhausted */

--- a/nczarr_test/ut_chunking.c
+++ b/nczarr_test/ut_chunking.c
@@ -3,8 +3,6 @@
  *      See netcdf/COPYRIGHT file for copying and redistribution conditions.
  */
 
-#define ZUT
-
 #include "ut_includes.h"
 
 void ut_chunk_print(int sort, ...);
@@ -29,8 +27,9 @@ main(int argc, char** argv)
     if((stat = ut_init(argc, argv, &utoptions))) goto done;
 
     /* Set the printer */
-    zutest.tests = UTEST_RANGE;
-    zutest.print = ut_chunk_print;
+    zutester.tests = UTEST_RANGE;
+    zutester.print = ut_chunk_print;
+    zutest = &zutester;
 
     var = nclistget(utoptions.vardefs,0);
 

--- a/nczarr_test/ut_includes.h
+++ b/nczarr_test/ut_includes.h
@@ -25,4 +25,6 @@
 #include "ut_test.h"
 #include "ut_util.h"
 
+static struct ZUTEST zutester;
+
 #endif /*UT_INCLUDES_H*/

--- a/nczarr_test/ut_projections.c
+++ b/nczarr_test/ut_projections.c
@@ -27,8 +27,9 @@ main(int argc, char** argv)
     if((stat = ut_init(argc, argv, &utoptions))) goto done;
 
     /* printer off for these tests */
-    zutest.tests = 0;
-    zutest.print = NULL;
+    zutester.tests = 0;
+    zutester.print = NULL;
+    zutest = &zutester;
 
     var = nclistget(utoptions.vardefs,0);
 

